django/core/files/storage/filesystem.py

```diff
@@ -98,12 +96,29 @@ def _save(self, name, content):
             try:
                 # This file has a file path that we can move.
                 if hasattr(content, "temporary_file_path"):
-                    file_move_safe(content.temporary_file_path(), full_path)
+                    file_move_safe(
+                        content.temporary_file_path(),
+                        full_path,
+                        allow_overwrite=self.ALLOW_OVERWRITE,
+                    )
 
                 # This is a normal uploadedfile that we can stream.
                 else:
+                    # The combination of O_CREAT and O_EXCL makes os.open() raise
+                    # OSError if the file already exists before it's opened.
+                    OS_OPEN_FLAGS = (
+                        os.O_WRONLY
+                        | os.O_CREAT
+                        | os.O_EXCL
+                        | getattr(os, "O_BINARY", 0)
+                    )
+
+                    if self.ALLOW_OVERWRITE:
+                        open_flags = OS_OPEN_FLAGS & ~os.O_EXCL
+                    else:
+                        open_flags = OS_OPEN_FLAGS
```

Comment:
I haven’t understood the problem hard enough to have an opinion about this change, but this pattern would be simpler:

```suggestion
                    open_flags = (
                        os.O_WRONLY
                        | os.O_CREAT
                        | getattr(os, "O_BINARY", 0)
                    )

                    if not self.ALLOW_OVERWRITE:
                        open_flags |= os.O_EXCL
```

Reply:
Thanks, I updated the PR.